VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -1,132 +1,126 @@
-
-
-                Intent resultIntent = new Intent(context, CourseDetailTabActivity.class);
-                resultIntent.setAction(Long.toString(System.currentTimeMillis()));
-                //if user launch the app from recent list, activity will still get this intent.
-                //this is one way to avoid it.
-                resultIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK |Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
-                resultIntent.putExtra( Router.EXTRA_BUNDLE, courseBundle);
-
-                // Because clicking the notification opens a new ("special") activity, there's
-                // no need to create an artificial back stack.
-                PendingIntent resultPendingIntent =  PendingIntent.getActivity(
-                        context,
-                        0,
-                        resultIntent,
-                        PendingIntent.FLAG_UPDATE_CURRENT
-                );
-
-                NotificationManager notificationManager =
-                        (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
-
-                NotificationCompat.Builder builder = new NotificationCompat.Builder(context);
-                builder.setContentTitle(titleTemplate);
-                builder.setContentText(payload.getCourseName());
-                builder.setSmallIcon(R.drawable.app_icon);
-                builder.setContentIntent(resultPendingIntent);
-                builder.setAutoCancel(true);
-
-                Uri uri= RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
-                builder.setSound(uri);
-
-                Notification notification = builder.build();
-                notification.flags |= Notification.FLAG_AUTO_CANCEL;
-                return notification;
-            } else {
-                EventBus.getDefault().postSticky(new CourseAnnouncementEvent(
-                    CourseAnnouncementEvent.EventType.MESSAGE_RECEIVED, ""));
-                return super.getNotification(context,intent);
+    @Override
+    protected android.app.Notification getNotification(@NonNull Context context, @NonNull Intent intent) {
+        final BaseNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
+        if (null != payload && null != payload.getAction()) {
+            switch (payload.getAction()) {
+                case ParseHandleHelper.COURSE_ANNOUNCEMENT_ACTION: {
+                    final CourseUpdateNotificationPayload courseUpdateNotificationPayload = (CourseUpdateNotificationPayload) payload;
+                    getSegment(context).trackNotificationReceived(courseUpdateNotificationPayload.getCourseId());
+                    return newNotificationBuilder(context, intent)
+                            .setContentTitle(context.getString(R.string.COURSE_ANNOUNCEMENT_NOTIFICATION_TITLE))
+                            .setContentText(courseUpdateNotificationPayload.getCourseName())
+                            .build();
```

Comment:
If we're only going to customize the title and text, then we could change the `newNotificationBuilder()` to return a `Notification` while accepting these two parameters. On the other hand, if we're anticipating arbitrary customizations (icon, sound, etc.), then it's fine as it is.